pkg/services/ngalert/provisioning/contactpoints.go

```diff
@@ -0,0 +1,338 @@
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/services/secrets"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/prometheus/alertmanager/config"
+)
+
+type EmbeddedContactPointService struct {
+	amStore           store.AlertingStore
+	encryptionService secrets.Service
+	provenanceStore   ProvisioningStore
+	xact              TransactionManager
+	log               log.Logger
+}
+
+func NewEmbeddedContactPointService(store store.AlertingStore, encryptionService secrets.Service,
+	provenanceStore ProvisioningStore, xact TransactionManager, log log.Logger) *EmbeddedContactPointService {
+	return &EmbeddedContactPointService{
+		amStore:           store,
+		encryptionService: encryptionService,
+		provenanceStore:   provenanceStore,
+		xact:              xact,
+		log:               log,
+	}
+}
+
+func (ecp *EmbeddedContactPointService) GetContactPoints(ctx context.Context, orgID int64) ([]apimodels.EmbeddedContactPoint, error) {
+	cfg, _, err := ecp.getCurrentConfig(ctx, orgID)
+	if err != nil {
+		return nil, err
+	}
+	provenances, err := ecp.provenanceStore.GetProvenances(ctx, orgID, "contactpoint")
+	if err != nil {
+		return nil, err
+	}
+	ecp.log.Info("provenances", "map", provenances)
+	contactPoints := []apimodels.EmbeddedContactPoint{}
+	for _, contactPoint := range cfg.GetGrafanaReceiverMap() {
+		embeddedContactPoint := apimodels.EmbeddedContactPoint{
+			UID:                   contactPoint.UID,
+			Type:                  contactPoint.Type,
+			Name:                  contactPoint.Name,
+			DisableResolveMessage: contactPoint.DisableResolveMessage,
+			Settings:              contactPoint.Settings,
+		}
+		if val, exists := provenances[embeddedContactPoint.UID]; exists && val != "" {
+			embeddedContactPoint.Provenance = string(val)
+		}
+		for k, v := range contactPoint.SecureSettings {
+			decryptedValue, err := ecp.decrypteValue(v)
+			if err != nil {
+				// TODO(JP): log a warning
+				continue
+			}
+			if decryptedValue == "" {
+				continue
+			}
+			embeddedContactPoint.Settings.Set(k, apimodels.RedactedValue)
+		}
+		contactPoints = append(contactPoints, embeddedContactPoint)
+	}
+	return contactPoints, nil
+}
+
+// internal only
+func (ecp *EmbeddedContactPointService) getContactPointUncrypted(ctx context.Context, orgID int64, uid string) (apimodels.EmbeddedContactPoint, error) {
+	cfg, _, err := ecp.getCurrentConfig(ctx, orgID)
+	if err != nil {
+		return apimodels.EmbeddedContactPoint{}, err
+	}
+	for _, receiver := range cfg.GetGrafanaReceiverMap() {
+		if receiver.UID != uid {
+			continue
+		}
+		embeddedContactPoint := apimodels.EmbeddedContactPoint{
+			UID:                   receiver.UID,
+			Type:                  receiver.Type,
+			Name:                  receiver.Name,
+			DisableResolveMessage: receiver.DisableResolveMessage,
+			Settings:              receiver.Settings,
+		}
+		for k, v := range receiver.SecureSettings {
+			decryptedValue, err := ecp.decrypteValue(v)
```

Comment:
Not sure if this PR added it, but it should be `decryptValue()`

Reply:
fixed